spec/02-integration/04-admin_api/08-targets_routes_spec.lua

```diff
@@ -302,24 +302,26 @@ describe("Admin API", function()
       end)
 
       it("only shows active targets", function()
-        local res = assert(client:send {
-          method = "GET",
-          path = "/upstreams/" .. upstream_name3 .. "/targets/active/",
-        })
-        assert.response(res).has.status(200)
-        local json = assert.response(res).has.jsonbody()
+        for _, append in ipairs({ "", "/" }) do
```

Comment:
Not a blocker but fwiw, I don't think this is a healthy way of writing tests. If one behavior fails (with or without trailing slash), the developer sees a test named "only shows active targets" fail, which isn't what the test was actually asserting. On top of that, the developer has no easy way of knowing which of the two tests (the one with or without the trailing slash) failed, which is confusing and/or frustrating.

Tests must be atomic and explicit, always. "Meta-programming" testing (generating tests based on loops) should be avoided when it breaks the atomicity of the test.